src/System.Collections.NonGeneric/tests/Performance/Perf.ArrayList.cs

```diff
@@ -0,0 +1,476 @@
+            {
+                ArrayList[] elementsCollection = new ArrayList[size];
+                for (int index = 0; index < size; index++)
+                {
+                    elementsCollection[index] = new ArrayList(elements);
+                }
+
+                using (iteration.StartMeasurement())
+                {
+                    for (int index = 0; index < innerIterationCount; index++)
+                    {
+                        elementsCollection[index].RemoveRange(1, size - 2);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 1000)]
+        public void Reverse()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            int startIndex = 0;
+            int count = 100000;
+            ArrayList elements = CreateArrayListOfInts(count);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < innerIterationCount; i++)
+                    {
+                        elements.Reverse(startIndex, count);
+                    }                   
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 10000000)]
+        public void SetRange()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            int startIndex = 0;
+            int size = 100000;
+            ArrayList elements = CreateArrayListOfInts(size);
+            var newElemnts = new ArrayList(size);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < innerIterationCount; i++)
+                    {
+                        elements.SetRange(startIndex, elements);
+                    }                    
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 20000000)]
+        public void GetRange()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            int startIndex = 0;
+            int size = 100000;
+            ArrayList elements = CreateArrayListOfInts(size);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < innerIterationCount; i++)
+                    {
+                        elements.GetRange(startIndex, size);
+                    }                    
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 10000000)]
+        public void Sort()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            int size = (int)Benchmark.InnerIterationCount;           
+            var random = new Random();
```

Comment:
The randomness here can affect the timing, since the values used can affect how many comparison operations are needed in the sort and thus the running time of the sort.  It'd be best to use a fixed seed with the Random ctor, so that all runs employ the same values and are deterministic.

Reply:
Is there value in having a benchmark for a "pathological" input, similar to what you fixed in Linq recently? Eg., sorted input. I'm guessing probably not, since it would only matter if someone changed the sort algorithm, and they're not going to do that without measuring..

Reply:
ArrayList.Sort just uses Array.Sort.  We should have a variety of tests for Array.Sort, but I think this is sufficient for ArrayList.Sort.

Reply:
This is fixed too...